src/WebJobs.Script.WebHost/Controllers/FunctionsController.cs

```diff
@@ -83,8 +83,10 @@ public FunctionsController(WebScriptHostManager scriptHostManager, WebHookReceiv
             }
             else
             {
+                var settings = (WebHostSettings)controllerContext.Configuration.DependencyResolver.GetService(typeof(WebHostSettings));
+
                 // Authorize
-                if (authorizationLevel < httpFunctionMetadata.AuthLevel)
+                if (authorizationLevel < httpFunctionMetadata.AuthLevel && !settings.IsSelfHost)
```

Comment:
Is your intent to bypass auth for all local requests? Then perhaps you should set authorizationLevel to Admin above when we determine it. As it stands now, you'll still be doing auth checks on WebHooks